src/System.Memory/tests/SequenceReader/Rewind.cs

```diff
@@ -67,5 +67,18 @@ public void Rewind_ByOne()
                 Assert.True(copy.CurrentSpan.SequenceEqual(reader.CurrentSpan));
             }
         }
+
+        [Fact]
+        public void Rewind_Exception()
+        {
+            ReadOnlySequence<byte> bytes = SequenceFactory.Create(new byte[][] {
+                new byte[] { 0          },
+                new byte[] { 1, 2       },
+                new byte[] { 3, 4       },
+                new byte[] { 5, 6, 7, 8 }
+            });
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => new SequenceReader<byte>(bytes).Rewind(-1));
```

Comment:
Similarly, can you add an assert where you advance by some value x, and then rewind by x + 1 (which should result in OutOfRange Exception as well).